src/main/java/seedu/address/model/schedule/Event.java

```diff
@@ -102,13 +115,26 @@ public boolean equals(Object other) {
         return otherEvent.getEventDescription().equals(getEventDescription())
                 && otherEvent.getDate().equals(getDate())
                 && otherEvent.getTime().equals(getTime())
-                && otherEvent.getDuration().equals(getDuration());
+                && otherEvent.getDuration().equals(getDuration())
+                && otherEvent.getRecurFrequency().equals(getRecurFrequency());
     }
 
     @Override
     public String toString() {
-        return String.format("%s %s %s-%s%s", eventDescription,
-                date.format(DateTimeFormatter.ofPattern("dd-MMM-YYYY")), time, getEndTime(), getDurationDays());
+        String plusDays = "";
+        long numDays = ChronoUnit.DAYS.between(getDate(), getEndDate());
+        if (numDays > 0) {
+            plusDays = String.format(" (+%s)", numDays);
+        }
+
+        if (!isRecurring()) {
+            return String.format("%s %s %s-%s%s", eventDescription,
+                    date.format(DateTimeFormatter.ofPattern("dd-MMM-YYYY")), time, getEndTime(), plusDays);
+        } else {
+            return String.format("%s %s %s-%s%s %s", eventDescription, date.format(
+                    DateTimeFormatter.ofPattern("dd-MMM-YYYY")), time, getEndTime(),
+                    plusDays, recurFrequency.toString());
+        }
```

Comment:
can just 
return String.format("%s %s %s-%s%s%s", eventDescription, date.format(
                      DateTimeFormatter.ofPattern("dd-MMM-YYYY")), time, getEndTime(),
                      plusDays, getRecurFrequency().map(x -> " " + x.toString()).orElse(""));

Reply:
Great suggestion! I will update that.